scripts/add_extra_components.py

```diff
@@ -185,6 +185,40 @@ def attach_stores(n, costs, config):
             marginal_cost=costs.at["battery inverter", "marginal_cost"],
         )
 
+    if "csp" in config["renewable"].keys():
+        if (
+            config["renewable"]["csp"]["architecture"] == "advanced"
+            and "csp" in carriers
+        ):
+            csp_buses_i = n.madd("Bus", buses_i + " csp", carrier="csp", **bus_sub_dict)
+
+            # change bus of existing csp generators
+            old_csp_bus_vector = buses_i + " csp"
+            n.generators.loc[old_csp_bus_vector, "bus"] = csp_buses_i
```

Comment:
As we discussed, using `n.madd` output as an index is an elegant solution under the current implementation of PyPSA, but is risky in long-term perspective. I'd suggest to better use an index of `csp_buses_i` explicitly.